lib/web_ui/test/golden_tests/engine/canvas_golden_test.dart

```diff
@@ -105,8 +113,61 @@ void main() async {
       ..color = const Color.fromRGBO(0, 255, 0, 1.0)
       ..style = PaintingStyle.fill);
 
-    html.document.body.append(canvas.rootElement);
+    appendToScene();
 
     await matchGoldenFile('bitmap_canvas_fills_paint_when_transformed.png', region: region);
   }, timeout: const Timeout(Duration(seconds: 10)));
+
+  // This test reproduces text blurriness when two pieces of text appear inside
+  // two nested clips:
+  //
+  //   ┌───────────────────────┐
+  //   │   text in outer clip  │
+  //   │ ┌────────────────────┐│
+  //   │ │ text in inner clip ││
+  //   │ └────────────────────┘│
+  //   └───────────────────────┘
+  //
+  // This test clips using canvas. See a similar test in `compositing_golden_test.dart`,
+  // which clips using layers.
+  //
+  // More details: https://github.com/flutter/flutter/issues/32274
+  test('renders clipped DOM text with high quality', () async {
```

Comment:
Do you think we should add a test for `devicePixelRatio > 1.0`?

Reply:
Great idea to test on `devicePixelRatio > 1.0` screens. However, I imagine instead of writing per-devicePixelRatio tests, a more effective strategy is to have a test agent with `devicePixelRatio > 1.0` that runs _all tests_. We use it many other places too.

I filed https://github.com/flutter/flutter/issues/48898

Reply:
I like the idea of having a test agent with `dpr>1.0` run all tests. But I think that's a separate idea that won't help your test in this PR. In fact, your test would break if it runs with a `dpr>1.0`.

IMHO, there are cases (like this PR) where you want to control the `dpr` value for a test and have different expectations based on the `dpr`.

Reply:
Discussed this offline. I'll leave it as is and bank on availability of high DPR infra later this quarter.